Sources/XCBuildSupport/XcodeBuildSystem.swift

```diff
@@ -171,9 +171,19 @@ public final class XcodeBuildSystem: SPMBuildCore.BuildSystem {
         // Always specify the path of the effective Swift compiler, which was determined in the same way as for the native build system.
         settings["SWIFT_EXEC"] = buildParameters.toolchain.swiftCompiler.pathString
         settings["LIBRARY_SEARCH_PATHS"] = "$(inherited) \(buildParameters.toolchain.toolchainLibDir.pathString)"
-        settings["OTHER_CFLAGS"] = "$(inherited) \(buildParameters.toolchain.extraCCFlags.joined(separator: " "))"
-        settings["OTHER_CPLUSPLUSFLAGS"] = "$(inherited) \(buildParameters.toolchain.extraCPPFlags.joined(separator: " "))"
-        settings["OTHER_SWIFT_FLAGS"] = "$(inherited) \(buildParameters.toolchain.extraSwiftCFlags.joined(separator: " "))"
+        settings["OTHER_CFLAGS"] = (
+            ["$(inherited)"] + buildParameters.toolchain.extraCCFlags + buildParameters.flags.cCompilerFlags
+        ).joined(separator: " ")
```

Comment:
I know this is existing code, but it would probably be good to do `.map{ $0.spm_shellEscaped() }.joined(separator: " ")` here.  Build setting quoting and shell quoting aren't exactly the same but this should be more conservative and avoid breaking on things like space characters in paths.

Reply:
It actually failed the tests to do shell escaping on the entire array, but I added it to just the inputs that are coming from the command line instead and it works as expected. thanks!

Reply:
Hmm, that's odd — I'll make a note to take a look at that.  Either way, this looks good.  Thanks!